tests/testthat/helper-initialize.R

```diff
@@ -18,3 +30,41 @@ sklearn_iris_dataset <- sklearn$datasets$load_iris()
 sort_mat <- function(m, cols = seq(ncol(m))) {
   m[do.call(order, lapply(cols, function(x) m[, x])), ]
 }
+
+#' Attempt to unserialize a CuML model within a sub-process and use the
+#' unserialized model to make predictions.
+predict_in_sub_proc <- function(model_state, data, expected_mode,
+                                expected_model_cls = NULL,
+                                additional_predict_args = list()) {
+  impl <- function(expect_libcuml_impl, model_state, data, expected_mode,
+                   expected_model_cls, additional_predict_args) {
+    library(cuml)
+    expect_libcuml_impl()
+
+    model <- cuml_unserialize(model_state)
+    missing_cls <- setdiff(expected_model_cls, class(model))
+    if (length(missing_cls) > 0) {
+      stop(
+        "Unserialized model is missing the following classes:\b",
+        paste(missing_cls, collapse = " "),
+        "\n"
+      )
+    }
+    stopifnot(identical(model$mode, expected_mode))
+
+    do.call(predict, append(list(model, data), additional_predict_args))
+  }
+
+  callr::r(
+    impl,
+    args = list(
+      expect_libcuml_impl = expect_libcuml,
```

Comment:
This is OK because `saveRDS()` (used by `callr` to serialize args) won't ever bind symbols within the function body of `expect_libcuml()` to anything within the current process.

Reply:
I think it's fine.